src/tests/ftest/util/results_utils.py

```diff
@@ -0,0 +1,331 @@
+            uid (str): the test uid
+            variant (str): the test variant
+        """
+        self.name = name
+        self.uid = uid
+        self.variant = variant
+
+    def __str__(self):
+        """Get the test name as a string.
+
+        Returns:
+            str: the test name including the uid, name, and variant
+
+        """
+        return "-".join([str(self.uid), "".join([str(self.name), str(self.variant)])])
+
+
+class TestResult():
+    """Provides the necessary test result data to generate a xml/html results file."""
+
+    PASS = "PASS"
+    WARN = "WARN"
+    SKIP = "SKIP"
+    FAIL = "FAIL"
+    ERROR = "ERROR"
+    CANCEL = "CANCEL"
+    INTERRUPT = "INTERRUPT"
+
+    def __init__(self, class_name, name, log_file, log_dir):
+        """Initialize a TestResult object.
+
+        Args:
+            class_name (str): the test class name, e.g. FTEST_<name>
+            name (TestName): the test uid, name, and variant
+            uid (str): the test uid
+            variant (str): the test variant
+            log_file (str): the log file for a single test
+            log_dir (str): the log file directory for a single test
+        """
+        self.class_name = class_name
+        self.name = name
+        self.logfile = log_file
+        self.logdir = log_dir
+        self.time_start = -1
+        self.time_end = -1
+        self.time_elapsed = -1
+        self.status = None
+        self.fail_class = None
+        self.fail_reason = None
+        self.traceback = None
+
+    def __getitem__(self, name, default=None):
+        """Get the value of the attribute name.
+
+        Args:
+            name (str): name of the class attribute to get
+            default (object, optional): value to return if name is not defined. Defaults to None.
+
+        Returns:
+            object: the attribute value or default if not defined
+
+        """
+        return self.get(name, default)
+
+    def __contains__(self, item):
+        """Determine if the attribute exists in this class, e.g. 'item' in self.
+
+        This is used when creating the html test result.
+
+        Args:
+            item (str): attribute name
+
+        Returns:
+            bool: True if the attribute name has a value in this object; False otherwise
+
+        """
+        return self.get(item, None) is not None
+
+    def get(self, name, default=None):
+        """Get the value of the attribute name.
+
+        Args:
+            name (str): name of the class attribute to get
+            default (object, optional): value to return if name is not defined. Defaults to None.
+
+        Returns:
+            object: the attribute value or default if not defined
+
+        """
+        try:
+            return getattr(self, name, default)
+        except TypeError:
+            return default
+
+    def start(self):
+        """Mark the start of the test."""
+        if self.time_start == -1:
+            self.time_start = time.time()
```

Comment:
I think this should either always set, even `if != 0`, OR, it should raise an exception to avoid confusion